src/xpk/commands/storage.py

```diff
@@ -0,0 +1,83 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from argparse import Namespace
+
+from kubernetes import client as k8s_client
+from kubernetes.client.exceptions import ApiException
+
+from ..core.core import (
+    setup_k8s_env,
+    update_cluster_with_gcsfuse_driver_if_necessary,
+    update_cluster_with_workload_identity_if_necessary,
+)
+from ..core.storage import (
+    STORAGE_CRD_KIND,
+    XPK_API_GROUP_NAME,
+    XPK_API_GROUP_VERSION,
+    create_storage_instance,
+    get_storage,
+    install_storage_crd,
+    list_storages,
+    print_storages_for_cluster,
+)
+from ..utils import apply_kubectl_manifest, xpk_exit, xpk_print
+
+
+def storage_create(args: Namespace) -> None:
+  k8s_api_client = setup_k8s_env(args)
+
+  install_storage_crd(k8s_api_client)
+  return_code = update_cluster_with_workload_identity_if_necessary(args)
+  if return_code > 0:
+    xpk_exit(return_code)
+  return_code = update_cluster_with_gcsfuse_driver_if_necessary(args)
+  if return_code > 0:
+    xpk_exit(return_code)
+
+  create_storage_instance(k8s_api_client, args)
+  apply_kubectl_manifest(k8s_api_client, args.manifest)
+
+
+def storage_list(args: Namespace) -> None:
+  k8s_api_client = setup_k8s_env(args)
+  install_storage_crd(k8s_api_client)
+  storages = list_storages(k8s_api_client)
+  print_storages_for_cluster(storages, args.cluster)
+
+
+def storage_delete(args: Namespace) -> None:
+  k8s_api_client = setup_k8s_env(args)
+  install_storage_crd(k8s_api_client)
+  api_instance = k8s_client.CustomObjectsApi(k8s_api_client)
+  core_api = k8s_client.CoreV1Api()
+  try:
+    storage = get_storage(k8s_api_client, args.name)
+    core_api.delete_namespaced_persistent_volume_claim(storage.pvc, "default")
```

Comment:
The order or deletion is correct now. 
- Is it guaranteed that if any failure happens during  the deletion of this objects we will get an exception? I'm not sure about this particular case, but for example when you "kubectl apply -f aDeployment.yaml", you get the okay that the request is sent, but the deployment may fail while the controller is provisioning the deployment.

- Let's say pvc and pv were deleted successfully, but custom_object failed. What's the behavior when retrying to delete the already deleted pv and pvc?

Reply:
> Is it guaranteed that if any failure happens during the deletion of this objects we will get an exception? I'm not sure about this particular case, but for example when you "kubectl apply -f aDeployment.yaml", you get the okay that the request is sent, but the deployment may fail while the controller is provisioning the deployment.

I couldn't find an exhausting list of Exceptions that can be throw by any of these methods, but by looking at examples provided in the k8s python client documentation I deduced they can only throw ApiExceptions

References:
- https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#delete_persistent_volume
- https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#delete_namespaced_persistent_volume_claim
- https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CustomObjectsApi.md#delete_cluster_custom_object

> Let's say pvc and pv were deleted successfully, but custom_object failed. What's the behavior when retrying to delete the already deleted pv and pvc?

Good catch. Currently if we tried to retry we would get 404 error and the storage wouldn't be deleted. Let me fix that

Reply:
Done, PTAL @sharabiani